File: src/models/Carte.ts
import { Commentaire } from "./Commentaire";
import { Projet } from "./Projet";

export class Carte {
  idCarte?: number;
  titre?: string;
  description?: string;
 
 
  projet!: Projet | null; ; // Assuming Projet interface matches the structure
  commentaires!: Commentaire[]; // Assuming Commentaire interface matches the structure
}

File: src/main/java/com/example/gestionprojeet/classes/Carte.java
package com.example.gestionprojeet.classes;

import jakarta.persistence.*;
import lombok.*;

import java.util.Date;
import java.util.List;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class Carte {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int idCarte;

    private String titre;
    private String description;
    private Date dateCreation;
    private Date dateEcheance;
    @ManyToOne

    private Projet projet;
    @OneToMany(mappedBy = "carte")
    private List<Commentaire> commentaires;
}
